### PR TITLE
Milestone 10 : Local Storage

### PR DESCRIPTION
**Pull request by Zohra-Neda**
**In this project we :**
- Preserved form data in the browser by using local storage

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -174,18 +174,35 @@ window.addEventListener('load', popUpWindow);
...
 
+// To retrieve data from local storage
+const storedData = localStorage.getItem('formData');
+if (storedData) {
+  const parsedData = JSON.parse(storedData);
+  username.value = parsedData.username;
+  email.value = parsedData.email;
+  textArea.value = parsedData.msg;
+}
+
```
Comment (aamirkhan2478) [inline script.js:207]: - [ ] Good effort :+1: but in [requirement](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md): When the user changes the input field's content, the data is saved to the local storage but in your case when I submit the form then it stored in the local storage so kindly fix this issue :slightly_smiling_face:

File: index.html
Comment (aamirkhan2478) [inline index.html:1]: [OPTIONAL] This requirement is optional but it is recommended please stick to the Figma design your design is  not the same as the Figma design check below :point_down: the comparison between your design and Figma design :slightly_smiling_face: 

**Your Design**
Screenshot from 2023-04-20 18-09-32

**Figma Design**
Screenshot from 2023-04-20 18-09-47

- If you check your button it does not align properly so kindly fix this issue also :+1: 
- Also, check your textarea's placeholder it's not the same as the Figma design so kindly fix the padding of your textarea :+1:

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @Zohra-Neda & @riskmaris, 



Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Good Effort
✔️ Your design is good.
✔️ All linter passed

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DuaneDave — APPROVED** [review]
Hi @Zohra-Neda 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Data is correctly saved to local storage
✔️ Data is correctly pull and all fields are pre-populated with it

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**